web/impact/impact/v1/events/base_user_role_grant_event.py

```diff
@@ -0,0 +1,52 @@
+from abc import (
+    ABCMeta,
+    abstractmethod,
+)
+from datetime import datetime
+from impact.models import ProgramRoleGrant
+from impact.utils import compose_filter
+
+
+class BaseUserRoleGrantEvent(object):
+    __metaclass__ = ABCMeta
+
+    def __init__(self, program_role_grant):
+        self.program_role_grant = program_role_grant
+
+    @abstractmethod
+    def description(self):
+        pass  # pragma: no cover
+
+    @classmethod
+    def events(cls, user):
+        result = []
+        user_role_name_filter = compose_filter(
+            ["program_role", "user_role", "name"],
+            cls.USER_ROLE)
+        for prg in user.programrolegrant_set.filter(**user_role_name_filter):
+            result.append(cls(prg))
+        return result
+
+    def serialize(self):
+        earliest, latest = self._user_event_datetime()
+        return {
+            "datetime": earliest,
+            "latest_datetime": latest,
```

Comment:
its not 100% clear what the 'latest_datetime' is supposed to note - only after looking through the code did i understand what this was. Given that the history object is not a model I suggest the options call returns something like the following which lists the parameters to expect (since its not related to an object that is self-describing):

```
{
    "name": "Organization List",
    "description": "",
    "renders": [
        "application/json",
        "text/html"
    ],
    "parses": [
        "application/json",
        "application/x-www-form-urlencoded",
        "multipart/form-data"
    ],
    "actions": {
        "GET": {
               "latest_datetime": {
                    "type": "string",
                   "description": "the latest update/action of the entity"
               },
                "datetime":{
                   "type": "string",
                   "description": "the earliest time this object was updated"
              },
     }
}
...

```